cclient/api/extended/get_inputs.c

```diff
@@ -13,51 +13,60 @@
 retcode_t iota_client_get_inputs(iota_client_service_t const* const serv, flex_trit_t const* const seed,
                                  address_opt_t const addr_opt, uint64_t const threshold, inputs_t* const out_inputs) {
   retcode_t ret_code = RC_ERROR;
-  hash243_queue_entry_t* q_iter = NULL;
-  size_t counter = 0;
-  get_balances_req_t* balances_req = get_balances_req_new();
-  get_balances_res_t* balances_res = get_balances_res_new();
-
-  log_debug(client_extended_logger_id, "[%s:%d]\n", __func__, __LINE__);
-  if (!balances_req || !balances_res) {
-    ret_code = RC_OOM;
-    log_error(client_extended_logger_id, "%s create get balances request or response object failed: %s\n", __func__,
-              error_2_string(ret_code));
-    return ret_code;
-  }
+  get_balances_req_t* balances_req = NULL;
+  get_balances_res_t* balances_res = NULL;
 
   out_inputs->total_balance = 0;
 
   // get address list
-  ret_code = iota_client_get_new_address(serv, seed, addr_opt, &balances_req->addresses);
-  if (ret_code) {
-    log_error(client_extended_logger_id, "%s get new address failed: %s\n", __func__, error_2_string(ret_code));
-    goto done;
-  }
+  address_opt_t addr_opt_elt = {addr_opt.start, 1, addr_opt.security};
+  for (uint64_t addr_index = addr_opt.start; addr_index < addr_opt.total; addr_index++) {
+    balances_req = get_balances_req_new();
+    balances_res = get_balances_res_new();
+    log_debug(client_extended_logger_id, "[%s:%d]\n", __func__, __LINE__);
+    if (!balances_req || !balances_res) {
+      ret_code = RC_OOM;
+      log_error(client_extended_logger_id, "%s create get balances request or response object failed: %s\n", __func__,
+                error_2_string(ret_code));
+      return ret_code;
+    }
+
+    ret_code = iota_client_get_new_address(serv, seed, addr_opt_elt, &balances_req->addresses);
```

Comment:
how about using `iota_sign_address_gen_flex_trits` instead of `iota_client_get_new_address`?

Reply:
In this new code, I used it.